stix_shifter/stix_transmission/src/modules/guardium/guardiumapiclient.py

```diff
@@ -0,0 +1,448 @@
+from ..utils.RestApiClient import RestApiClient
+from ..utils.RestApiClient import ResponseWrapper
+from requests.models import Response
+import urllib.parse
+import logging
+import sys
+import json
+import datetime
+import os
+import base64
+
+
+class APIClient():
+
+    # API METHODS
+
+    # These methods are used to call Ariel's API methods through http requests.
+    # Each method makes use of the http methods below to perform the requests.
+
+    # This class will encode any data or query parameters which will then be
+    # sent to the call_api() method of the RestApiClient
+    def __init__(self, connection, configuration):
+#
+        self.endpoint_start = 'restAPI/'
+        self.connection = connection
+        self.configuration = configuration
+        self.headers = dict()
+        self.search_id = None
+        self.query = None
+        self.authorization = None
+        self.credential = {}
+
+#
+# Check if connection object contains the following
+ 
+        username = connection.get("username", None)
+        password = connection.get("password", None)
+        grant_type = connection.get("grant_type", None)
```

Comment:
This and any other data-source-specific attributes may be coming in from the `connection.options`. So in this case it would be `connection.get('options', {}).get('grant_type')`. Also, the `.get` method will default to `None` if the key isn't found.

Reply:
Will username come from connection object ('options') or configuration object ('auth')?

Reply:
It will come from the auth object inside of the configuration

Reply:
If grant_type is always going to be "password" we will never include it in the connection object.